src/api/app/models/notification.rb

```diff
@@ -2,6 +2,9 @@ class Notification < ApplicationRecord
   belongs_to :subscriber, polymorphic: true
   belongs_to :notifiable, polymorphic: true
 
+  has_many :notified_projects, dependent: :destroy
+  has_many :projects, through: :notified_projects
```

Comment:
Testing, I tried to remove a notification and realized this raised an error. The problem seems to be that the `dependent: :destroy` should be included in the second line. Something like:

```
  has_many :notified_projects
  has_many :projects, through: :notified_projects, dependent: :destroy
```

This way it works fine and the record in the join table gets removed after a notification is removed.

Reply:
It's dropping also removing the `project` with your suggestion. I don't believe it's what we want. The current version removes the associated records in `notified_projects`. What is the error you got?

Reply:
This is the error:

```
  NotifiedProject Destroy (1.3ms)  DELETE FROM `notified_projects` WHERE `notified_projects`.`` IS NULL                                                                                       
   (0.6ms)  ROLLBACK                                                                                                                                                                          
Traceback (most recent call last):                                                                                                                                                            
        1: from (irb):8                                                                                                                                                                       
ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'notified_projects.' in 'where clause')
```

Reply:
Hum...we will need an ID because we're using a `has_many through:` association... See the docs [here](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-create_table).
```
:id

    Whether to automatically add a primary key column. Defaults to true. Join tables for ActiveRecord::Base.has_and_belongs_to_many should set it to false.

    A Symbol can be used to specify the type of the generated primary key column.
```

Reply:
I've adapted the migration and now it works as excepted.